### PR TITLE
tweak(subo): Rename scc -> scn in subo everywhere

### LINKED ISSUE #332: Rename SCC -> SCN everywhere
Sibling issue for https://github.com/suborbital/scn/issues/547

### PR DESCRIPTION
**Pull request by javorszky**
Closes #332 

Replaces #333 

- also fixed a variable / package name collision
- moved some details to consts, and updated usages accordingly

### CODE REVIEW

**Review by ospencer — COMMENTED** [review]
It looks like there are a lot of linting-related changes that aren't related to the SCC -> SCN swap. Should we land those separately?

File: templates/scc-docker/docker-compose.yml.tmpl [hunk 1]
```diff
@@ -1,41 +1,41 @@
 version: '3'
 services:
-  scc-control-plane:
-    image: suborbital/scc-control-plane:{{ .SCCVersion }}
+  scn-control-plane:
+    image: suborbital/scn-control-plane:{{ .SCNVersion }}
```
Comment (arbourd) [inline templates/scc-docker/docker-compose.yml.tmpl:4]:
```suggestion
  scn-controlplane:
    image: suborbital/scn-controlplane:{{ .SCNVersion }}
```

File: templates/scc-docker/docker-compose.yml.tmpl [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
       - scn
-    
-  scc-atmo:
+
+  scn-atmo:
     image: suborbital/atmo:v0.4.7
     command: atmo
     depends_on:
-      - scc-control-plane
+      - scn-control-plane
```
Comment (arbourd) [inline templates/scc-docker/docker-compose.yml.tmpl:36]:
```suggestion
      - scn-controlplane
```

File: templates/scc-docker/docker-compose.yml.tmpl [hunk 3]
```diff
@@ -1,41 +1,41 @@
...
     environment:
-      ATMO_CONTROL_PLANE: "scc-control-plane:8081"
+      ATMO_CONTROL_PLANE: "scn-control-plane:8081"
```
Comment (arbourd) [inline templates/scc-docker/docker-compose.yml.tmpl:38]:
```suggestion
      ATMO_CONTROL_PLANE: "scn-controlplane:8081"
```

File: templates/scc-k8s/.suborbital/scc-controlplane-deployment.yaml.tmpl [hunk 4]
```diff
@@ -2,99 +2,99 @@ apiVersion: apps/v1
...
     metadata:
       labels:
-        app: scc-controlplane
+        app: scn-controlplane
 
     spec:
       containers:
         - name: controlplane
-          image: suborbital/scc-control-plane:{{ .SCCVersion }}
+          image: suborbital/scn-control-plane:{{ .SCNVersion }}
```
Comment (arbourd) [inline templates/scc-k8s/.suborbital/scc-controlplane-deployment.yaml.tmpl:25]:
```suggestion
          image: suborbital/scn-controlplane:{{ .SCNVersion }}
```

**Review by arbourd — COMMENTED** [review]
Could we also normalize to `controlplane` over `control-plane`? This was always intended for the Docker images.

**Comment by javorszky** [comment]
> It looks like there are a lot of linting-related changes that aren't related to the SCC -> SCN swap. Should we land those separately?

@ospencer on one hand I understand the need for it, on the other, the linting was broken in the previous version as it used the linting config from a previous version of gci.

I figured if I have to change _some_ of the linting anyways, I might as well standardise the lints between subo and scn (gci and revive to unearth the import shadows).

To me it's a case of the boyscout rule (leave code looking better than you found it) in this case, and because it's somewhat related, I think it's fine to leave them in for now. Generally I try to separate out things into different PRs.

**Review by ospencer — APPROVED** [review]

**Review by arbourd — CHANGES_REQUESTED** [review]
Requires an SCC release

**Comment by arbourd** [comment]
Closing this as it has been done in `suborbital/templates`. Thanks @javorszky!

**Pull request closed without merging**